### PR TITLE
Added OpenLDAP chart

### PR DESCRIPTION
**Pull request by enis**
An initial helm chart for installing OpenLDAP server. 
It uses docker images from https://github.com/osixia/docker-openldap. Comes with some documentation and basic test. The chart is added to the incubator until it matures.

### CODE REVIEW

**Comment by viglesiasce** [comment]
@enis can you check the CLA?

**Comment by enis** [comment]
@viglesiasce thanks for checking. I've finally resolved the CCLA problems. It took some time.

**Review by unguiculus — CHANGES_REQUESTED** [review]
* Please make sure you have standard labels on all resouces. 
* The `app` label should be `{{ template "openldap.name" . }}`.
* Please update `_helpers.tpl` to the one created with `helm create` using Helm 2.8+.

**Comment by unguiculus** [comment]
ping @enis Any change you can update the PR?

**Comment by enis** [comment]
Sorry this fell from my radar. I have rebased the patch, and addressed the review comments. @unguiculus  do you mind taking a look. Thanks.

File: incubator/openldap/templates/openldap-customldif-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# /container/service/slapd/assets/config/bootstrap/ldif/custom
+#
+{{- if .Values.customLdifFiles }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "openldap.fullname" . }}-customldif-configmap
+  labels:
+    app: {{ template "openldap.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
```
Comment (unguiculus) [inline incubator/openldap/templates/openldap-customldif-configmap.yaml]: `chart: {{ template "openldap.chart" . }}`

Apply everywhere.

File: incubator/openldap/templates/openldap-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        app: {{ template "openldap.name" . }}
+        release: {{ .Release.Name }}
+    spec:
+      containers:
+      - name: {{ .Values.containerName | quote }}
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+        ports:
+        - name: ldap
+          containerPort: {{ .Values.service.ldapInternalPort }}
```
Comment (unguiculus) [inline incubator/openldap/templates/openldap-deployment.yaml]: It probably doesn't make sense to make container ports configurable. Change them will break things. I'd suggest you hard-code them, give them names, and reference them via their names from service and probes (which you should add).

File: incubator/openldap/templates/openldap-pvc.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+metadata:
+  name: {{ template "openldap.fullname" . }}
+  labels:
+    app: {{ template "openldap.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  accessModes:
+    - {{ .Values.persistence.accessMode | quote }}
```
Comment (unguiculus) [inline incubator/openldap/templates/openldap-pvc.yaml]: I'd like you to make list indentation consistent throughout the chart. Either use zero or two spaces.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/openldap/templates/openldap-service.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#
+# Service spec for openldap
+#
```
Comment (unguiculus) [inline incubator/openldap/templates/openldap-service.yaml]: Most of these comments are useless and just state the obvious. I'd remove them.

File: incubator/openldap/templates/tests/openldap-test-runner.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ template "openldap.fullname" . }}-service-test-{{ randAlphaNum 5 | lower }}"
```
Comment (unguiculus) [inline incubator/openldap/templates/tests/openldap-test-runner.yaml]: Add standard labels.

File: incubator/openldap/templates/tests/openldap-tests.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "openldap.fullname" . }}-tests
```
Comment (unguiculus) [inline incubator/openldap/templates/tests/openldap-tests.yaml]: Add standard labels.

File: incubator/openldap/templates/openldap-deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,88 @@
+#
+# A Deployment spec for openldap slapd
+#
+apiVersion: apps/v1beta1
```
Comment (unguiculus) [inline incubator/openldap/templates/openldap-deployment.yaml]: I'd suggest you use `apps/v1beta2`

File: incubator/openldap/README.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
+# OpenLDAP Helm Chart
+
+## Prerequisites Details
+* Kubernetes 1.6+
```
Comment (unguiculus) [inline incubator/openldap/README.md]: 1.8+

**Review by unguiculus — CHANGES_REQUESTED** [review]
Next time, please merge instead of rebasing, so Github can display the changes. I'd suggest you move the chart to stable.

File: incubator/openldap/templates/openldap-service.yaml (hunk 4 above)
Comment (enis) [reply incubator/openldap/templates/openldap-service.yaml]: Removed comments.

File: incubator/openldap/templates/tests/openldap-test-runner.yaml (hunk 5 above)
Comment (enis) [reply incubator/openldap/templates/tests/openldap-test-runner.yaml]: done.

File: incubator/openldap/templates/openldap-deployment.yaml (hunk 7 above)
Comment (enis) [reply incubator/openldap/templates/openldap-deployment.yaml]: done.

File: incubator/openldap/templates/tests/openldap-tests.yaml (hunk 6 above)
Comment (enis) [reply incubator/openldap/templates/tests/openldap-tests.yaml]: done.

File: incubator/openldap/README.md (hunk 8 above)
Comment (enis) [reply incubator/openldap/README.md]: fixed.

**Comment by enis** [comment]
Thanks @unguiculus I have addressed the review items.

File: stable/openldap/README.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+## Configuration
+
+We use the docker images provided by https://github.com/osixia/docker-openldap. The docker image is highly configurable and well documented. Please consult to documentation for the docker image for more information.
+
+The following tables lists the configurable parameters of the openldap chart and their default values.
+
+| Parameter                       | Description                           | Default                                                    |
+| ------------------------------- | ----------------------------------    | ---------------------------------------------------------- |
```
Comment (unguiculus) [inline stable/openldap/README.md]: Please format the table properly (align the pipes).

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/openldap/values.yaml [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+  LDAP_ORGANISATION: "Example Inc."
+  LDAP_DOMAIN: "example.org"
+  LDAP_BACKEND: "hdb"
+  LDAP_TLS: "true"
+  LDAP_TLS_ENFORCE: "false"
+  LDAP_REMOVE_CONFIG_AFTER_SETUP: "true"
+
+# Default Passwords to use. Unset these to auto-generate password stored as secrets.
+# You can override these at install time with
+# helm install openldap --set openldap.adminPassword=<passwd>,openldap.configPassword=<passwd>
```
Comment (unguiculus) [inline stable/openldap/values.yaml]: Don't set default passwords. Default should be auto-generation.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/openldap/values.yaml (hunk 10 above)
Comment (enis) [reply stable/openldap/values.yaml]: fixed.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**